add documentation about automating release using krew-release-bot

rajatjindal opened the pull request.

ahmetb commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 
+### Automating the releasing of new version
+
+You can use `github actions` to automate the release process. e.g. You can use
```
Don’t put github actions in code font?

ahmetb replied:
No need for e.g. here. You aren’t giving an example.

rajatjindal replied:
done

ahmetb commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
+### Automating the releasing of new version
+
+You can use `github actions` to automate the release process. e.g. You can use
+`go-releaser` (or equivalent for other languages) for creation of new release and 
```
Don’t mention goreleaser it is not relevant here.

rajatjindal replied:
done

ahmetb commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
 This helps users and maintainers to easily identify which version of the plugin
 they have installed.
 
+### Automating the releasing of new version
```
## Automate releasing new versions on krew-index

rajatjindal replied:
done

corneliusweig replied:
The latest version I see still says

> Automating the releasing of new version

@rajatjindal Can you check again?

rajatjindal replied:
yes, looks like I reverted that somehow, fixing again.

ahmetb commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
+You can use `github actions` to automate the release process. e.g. You can use
+`go-releaser` (or equivalent for other languages) for creation of new release and 
+[`krew-release-bot`](https://github.com/rajatjindal/krew-release-bot) to bump the 
+version in `krew-index` repo. Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot#basic-setup) for an example.
```
No need for #target link. 
Just link to your repo home. It should be concise enough for someone to understand how it works in 60 seconds or so. :)

rajatjindal replied:
done

ahmetb commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,10 @@ Ideally, the `version` specified should match the release tag of the plugin.
+### Automating the releasing of new version
+
+You can use github actions to publish new release of your krew plugin. `krew-release-bot`
+is a github action to automatically bump the version in `krew-index` repo. Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot) for details.
```
80 char line wraps pls :)

please capitalize as GitHub Actions.

rajatjindal replied:
sure, done

corneliusweig commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
+### Automating the releasing of new version
+
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a github action to automatically bump the version in 
```
```suggestion
`krew-release-bot` is a GitHub Action to automatically bump the version in 
```
Better be consistent with regards to capitalization. IMO, `Action` is a name here and needs to have a capital letter too.

rajatjindal replied:
done

corneliusweig commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a github action to automatically bump the version in 
+`krew-index` repo. 
+Refer to the [documentation here](https://github.com/rajatjindal/krew-release-bot) for details.
```
```suggestion
Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) documentation for details.
```

rajatjindal replied:
done

rajatjindal commented:
> First of all, thanks for pushing this forward @rajatjindal !
> 
> Before we merge this, let's get a common understanding about whether `krew-release-bot` should be the recommended procedure for updates, as stated on your project landing page. I'd prefer to not have this statement there, because the _real_ goal is to have PRs with regular changesets, so that the approver-bot can handle them. Also, it's not clear who recommends this.



> First of all, thanks for pushing this forward @rajatjindal !
> 
> Before we merge this, let's get a common understanding about whether `krew-release-bot` should be the recommended procedure for updates, as stated on your project landing page. I'd prefer to not have this statement there, because the _real_ goal is to have PRs with regular changesets, so that the approver-bot can handle them. Also, it's not clear who recommends this.


sorry for the confusion it has caused. My intention was not to show recommendation from krew team, rather that statement was added when krew-release-bot had two ways of doing PR. (webhook & github-action) and out of those two github-action was the recommended way of doing PR.

I will remove that statement to avoid any further confusion. thanks for pointing out.

corneliusweig commented on docs/DEVELOPER_GUIDE.md:
```diff
@@ -333,5 +333,12 @@ Ideally, the `version` specified should match the release tag of the plugin.
+You can use Github Actions to publish new release of your Krew plugin. 
+`krew-release-bot` is a Github Action to automatically bump the version in 
+`krew-index` repo. 
+Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) for details.
```
```suggestion
Refer to the [krew-release-bot](https://github.com/rajatjindal/krew-release-bot) documentation for details.
```

rajatjindal replied:
done

corneliusweig commented:
:smiley: no worries. I'm open to also make this a krew-recommended upgrade method for version bumps, but it's a little early for that. Let's first to gather some more experience.

ahmetb commented:
@corneliusweig tbh this is the "polished" experience compared to what it was before, and I suspect it's unlikely to change dramatically. I thought about image coming from a personal repo, but I think even that's OK for now. We just need to keep it static, as it works pretty decently at the moment.

corneliusweig commented:
As long as no credentials or access tokens are passed to the action, the image can be hosted anywhere.

@rajatjindal when you made your final change, let's merge this!

ahmetb commented:
Rather than the hosting of the image, the problem is to keep the API between container `<=>` service stable.

Since we don't use `:latest` in the image (and it would be dangerous to do so) any time we update the tag, we need to go update 20 other repos.

rajatjindal commented:
Hi @corneliusweig/@ahmetb  thanks for all the feedback to get the wording right. I think I've addressed all the review comments now. If there is anything else that needs to be changed, I will be more than happy to do so.

thanks

ahmetb commented:
Accidentally manually merged but
/lgtm
/approve
